### DISCUSSION TITLE
Mixed Kotlin/Java support?

### DISCUSSION THREAD

**Discussion by efenderbosch-atg** [mcarleio/konvert#92] [Q&A]
I'm currently use MapStruct, but would prefer a Kotlin native solution.

My API input classes are written in Java to avoid the "non-null parameter was null" exception that happens when deserializing new instances w/ Jackson. I use Jakarta/Hibernate validation to turn that in to a constraint violation instead, along with all the other input validations. However, my JPA entities (regular class) and API response models (data class) are Kotlin.

Can I map from a Java record class to a regular Kotlin class?

**Comment by efenderbosch-atg** [comment]
Did some playing around and it doesn't look like there's support for my use case.

Things missing:

meta-annotations: In MapStruct I can make reusable annotations like this:
```kotlin
@Retention(AnnotationRetention.SOURCE)
@Mapping(target = "id", ignore = true)
@Mapping(target = "createdBy", ignore = true)
@Mapping(target = "createdDate", ignore = true)
@Mapping(target = "updatedBy", ignore = true)
@Mapping(target = "updatedDate", ignore = true)
annotation class IgnoreJpaAttributes
```

mutating an existing object:
```kotlin
@IgnoreJpaAttributes
fun updateEntity(info: LotInfo, @MappingTarget entity: LotEntity)
```
The generated converter will take all matching attributes from `info` and mutate the `entity`.

**Reply by mcarleio** [reply]
I am currently on vacation. But in general, mapping between Java and Kotlin should work. There currently is indeed nothing like meta-annotations in Konvert. I will put it on the list of further ideas.

If you are interested in this, I am also open for a PR 💪
